Add 'notation' option to PluralRules tests and update related assertions

ryzokuken opened the pull request:
Refs: https://github.com/tc39/ecma402/pull/989

ptomato reviewed (commented):
Looks good, but on looking at the normative PR I think there should at least be a test that tries to construct a PluralRules with each valid value of `notation`, and verifies that the constructor throws on an invalid value such as `COMPACT` or `ståndard`.

ryzokuken commented:
@ptomato thanks for the review. I replaced the constructor test with what you suggested, moved the old test to `select` since it was testing that function's behavior anyway and edited another existing test that I thought could be further tested. I hope things look better now.

ptomato commented on test/intl402/PluralRules/prototype/constructor/notation.js:
```diff
@@ -0,0 +1,26 @@
+const invalidValues = ["COMPACT", "ståndard", 123, false, Symbol("foo"), null, {}, [], ""];
+
+for (const value of validValues) {
+  assert.doesNotThrow(() => {
```
`assert.doesNotThrow` is not a thing :smile: Either just `new Intl.PluralRules()` and do nothing with the created instance, or assert that it is of the correct type.

ptomato reviewed (approved):
Thanks, this looks like what I expected. Please do fix the assert function though.

eemeli commented on test/intl402/PluralRules/prototype/select/notation.js:
```diff
@@ -0,0 +1,20 @@
+    ...
+---*/
+
+assert.sameValue(new Intl.PluralRules('fr-FR', { notation: 'compact' }).select(1.00000020e6), 'one', 'compact notation');
```
Shouldn't this be `'many'`, given that the value formats the same as `1c6`?

eemeli replied:
Actually, the [CLDR plural rule](https://github.com/unicode-org/cldr/blob/faa2818490a34a2288f307720ab0abdab8c91ff7/common/supplemental/plurals.xml#L123-L127) examples for `many` even include `1.0000001c6`.

ryzokuken replied:
good point, I guess this applies to all related romance languages with a similar definition of `"many"` so the original example must've been unsound. Replacing it with a different language without that category like Slovene. Does that sound like a good idea?

trflynn89 commented on test/intl402/PluralRules/prototype/select/notation.js:
```diff
@@ -0,0 +1,20 @@
+    ...
+---*/
+
+assert.sameValue(new Intl.PluralRules('sl', { notation: 'compact' }).select(1.00000020e6), 'one', 'compact notation');
+assert.sameValue(new Intl.PluralRules('sl', { notation: 'standard' }).select(1.00000020e6), 'other', 'standard notation');
```
Curious about these values as well - trying to validate our implementation in Ladybird.

For `notation=standard`, since that's effectively the default value, wouldn't the result be "few"? Chrome, Firefox, Safari and Ladybird all agree on:
```js
>> new Intl.PluralRules('sl').select(1.00000020e6)
"few" 
```

and ICU continues to give us "few" after setting the specific notation.

ICU is also giving us "other" for `notation=compact`, regardless of whether I use a compact display of short or long.
